### DISCUSSION TITLE
[Help]: Cannot find module 'embla-carousel/components/Options' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by khlee0620** [davidjerleke/embla-carousel#726] [Help] [answered]
### Summary

I've installed 'shadcn/ui' in a Laravel Inertia + React environment and used it to set up 'embla-carousel-react'. However, after configuring, when I attempt a yarn build, I encounter an error. It seems there's an issue with a file within node_modules, and I'm unsure how to resolve this. Can you help?

`
node_modules/embla-carousel-react/esm/components/useEmblaCarousel.d.ts:9:39 - error TS2307: Cannot find module 'embla-carousel/components/Options' or its corresponding type declarations.

9     var globalOptions: Partial<import("embla-carousel/components/Options").OptionsType> | undefined;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/embla-carousel-react/esm/components/useEmblaCarousel.d.ts:9

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

_No response_

### CodeSandbox example

_No response_

**Comment by davidjerleke** [comment, marked as answer]
Hi @khlee0620,

Thank you for your question. Have you read [this](https://www.embla-carousel.com/api/options/#typescript) in the docs? It may be what you're looking for.

> [!NOTE]  
> Don't forget to switch to the `React` tab.

You didn't share any information about package versions and your setup so I'm just guessing here.

Best,
David

**Reply by khlee0620** [reply]
Thanks @davidjerleke, I'll try what you told me!

**Reply by khlee0620** [reply]
@davidjerleke , I tried the method you suggested, but encountered the same issue. However, I think I've found the cause elsewhere. The initial 'moduleResolution' setting in tsconfig.json was 'bundler', but changing it to 'node' seemed to fix the build issue. I still need to investigate the exact cause, but thank you for your help!

**Reply by davidjerleke** [reply]
@khlee0620 I see. I think you need to have `moduleResolution` set to `node` if you want to use CJS or ESM packages built with TypeScript. In other words I don’t think this is a problem with Embla Carousel.

Best,
David
